Workspaces/HierarchyWorkspace/Scripts/HierarchyListViewController.cs

```diff
@@ -41,6 +41,38 @@ sealed class HierarchyListViewController : NestedListViewController<HierarchyDat
 
 		readonly List<KeyValuePair<Transform, GameObject>> m_HoveredGameObjects = new List<KeyValuePair<Transform, GameObject>>();
 
+		public override List<HierarchyData> data
+		{
+			set
+			{
+				base.data = value;
+
+				if (m_Data != null && m_Data.Count > 0)
+				{
+					// Remove any objects that don't exist any more
+					var missingKeys = m_Data.Select(d => d.index).Except(m_ExpandStates.Keys);
```

Comment:
Hm. Fair point. I've been meaning to do another GC pass after this landed, so I'll see if this shows up on the profiler.